server/bin/pbench-index.py

```diff
@@ -181,344 +135,15 @@ def main(options, name):
-                    )
-                    with erred.open(mode="a") as fp:
-                        print(tb, file=fp)
-                    rename_tb_link(
-                        tb, Path(controller_path, f"{linkerrdest}.1"), idxctx.logger,
-                    )
-                elif tb_res in (2, 3):
-                    assert False, (
-                        f"Logic Bomb!  Unexpected tar ball handling "
-                        f"result status {tb_res:d} for tar ball {tb}"
-                    )
-                elif tb_res >= 4 or res <= 11:
-                    # # Quietly skip these errors
-                    with skipped.open(mode="a") as fp:
-                        print(tb, file=fp)
-                    rename_tb_link(
-                        tb,
-                        Path(controller_path, f"{linkerrdest}.{tb_res:d}"),
-                        idxctx.logger,
-                    )
-                else:
-                    idxctx.logger.error(
-                        "{}: index error {:d} encountered on {}", idxctx.TS, tb_res, tb
-                    )
-                    with erred.open(mode="a") as fp:
-                        print(tb, file=fp)
-                    rename_tb_link(
-                        tb, Path(controller_path, linkerrdest), idxctx.logger
-                    )
-                idxctx.logger.info("Finished {} (size {:d})", tb, size)
-        except Exception:
-            idxctx.logger.exception("Unexpected setup error")
-            res = 12
-        else:
-            # No exceptions while processing tar ball, success.
-            res = 0
-        finally:
-            if idxctx:
-                idxctx.dump_opctx()
-            idxctx.logger.debug("stopped processing list of tar balls")
-
-            idx = _count_lines(indexed)
-            skp = _count_lines(skipped)
-            err = _count_lines(erred)
-
-            idxctx.logger.info(
-                "{}.{}: indexed {:d} (skipped {:d}) results," " {:d} errors",
-                name,
-                idxctx.TS,
-                idx,
-                skp,
-                err,
-            )
-
-            if err > 0:
-                if skp > 0:
-                    subj = (
-                        f"{name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d}"
-                        f" results, w/ {err:d} errors"
-                    )
-                else:
-                    subj = (
-                        f"{name}.{idxctx.TS} - Indexed {idx:d} results, w/ {err:d}"
-                        " errors"
-                    )
-            else:
-                if skp > 0:
-                    subj = f"{name}.{idxctx.TS} - Indexed {idx:d} results, skipped {skp:d} results"
-                else:
-                    subj = f"{name}.{idxctx.TS} - Indexed {idx:d} results"
+    idx = Index(name, options, idxctx, INCOMING_rp)
+    status, value = idx.collect_tb(ARCHIVE_rp, qdir)
```

Comment:
The name `value` for the second element of the returned tuple seems a bit, shall we say, "under-descriptive".  ;). The return value of `collect_tb` appears to be a success/failure number, and a list of tar balls.  So why not rename `value` to be `tarballs`?

Reply:
I thought it will cause confusion if it will not be a list of tarballs. Like there are numeric return values 12 and 0

Reply:
When is `value` not a list of tar balls?

Reply:
That means it is an empty list of tarballs, still the same thing.  If we are returning `None` instead of an empty list, we should change that to an empty list.

Reply:
ok